app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -650,21 +674,32 @@ class BrowserActivity : DuckDuckGoActivity(), BookmarkDialogCreationListener, We
         viewModel.onFindResultsReceived(activeMatchOrdinal, numberOfMatches)
     }
 
+    private fun EditText.replaceTextChangedListener(textWatcher: TextChangedWatcher) {
+        removeTextChangedListener(textWatcher)
+        addTextChangedListener(textWatcher)
+    }
+
+    private fun hideKeyboard() {
+        omnibarTextInput.hideKeyboard()
+        focusDummy.requestFocus()
+    }
+
     private data class PendingFileDownload(
             val url: String,
             val directory: String
-        )
+    )
 
     companion object {
 
-        fun intent(context: Context, queryExtra: String? = null): Intent {
+        fun intent(context: Context, queryExtra: String? = null, replaceExistingSearch: Boolean = false): Intent {
             val intent = Intent(context, BrowserActivity::class.java)
-            intent.putExtra(QUERY_EXTRA, queryExtra)
+            intent.putExtra(EXTRA_TEXT, queryExtra)
+            intent.putExtra(REPLACE_EXISTING_SEARCH_EXTRA, replaceExistingSearch)
```

Comment:
>it feels like an internal detail to the browser to know what state it is in and whether it needs to clear itself

Externally, you know whether you want a browser activity which needs to be in a fresh state or not. You are specifically requesting it to come fresh.

>the only thing that needs it is the assistant

The widget also needs this functionality. The assist app and the widget are two places which want to launch a browser - but they want to guarantee a fresh view for it.

That said, you are right that it's visible to more classes than necessary. As such, I've removed some of the references from the `LandingActivity` as that will always result in a _fresh_ browser view anyway. 

>on a new intent

This is a good point; removed the processing of the intent from `onCreate` as we only care about `onNewIntent` flow.

Reply:
> Externally, you know whether you want a browser activity which needs to be in a fresh state or not
I see it differently - the external elements ask the Browser to launch or launch with a url. The Browser then uses this information with internal knowledge to decide whether it needs to rest anything to support this behavior. With new Intent I assume a reset of the activity, that we resuse the webview rather than completely reset it every time if we have a shared url seems an internal efficiency optimization.

I see where you're coming from too and am happy to go with the changes you've made. This will become moot for tabs anyway.